pages/api/ddd/v1/[ddd].js

```diff
@@ -3,12 +3,33 @@ import app from '@/app';
 import BaseError from '@/errors/BaseError';
 import InternalError from '@/errors/InternalError';
 import NotFoundError from '@/errors/NotFoundError';
+import BadRequestError from '@/errors/BadRequestError';
 
 import { getDddsData } from '@/services/ddd';
 
 async function citiesOfDdd(request, response, next) {
   try {
-    const requestedDdd = request.query.ddd;
+    let requestedDdd = request.query.ddd.toString();
+
+    const lengthDdd = requestedDdd.length;
+
+    if (lengthDdd === 0 || lengthDdd > 3) {
+      throw new NotFoundError({
+        message: 'DDD não encontrado',
+        type: 'ddd_error',
+        name: 'DDD_NOT_FOUND',
+      });
+    }
+
+    if (lengthDdd === 3 && requestedDdd[0] === '0') {
+      requestedDdd = requestedDdd.substring(1);
+    } else if (lengthDdd !== 2) {
+      throw new BadRequestError({
+        message: 'DDD inválido',
+        type: 'ddd_error',
+        name: 'DDD_INVALID',
+      });
+    }
```

Comment:
Uma maneira que julgo mais simples de resolver este problema é receber o valor, transformar em número e depois voltar para string. Ao fazer isto os zeros a esquerda são removidos.

```suggestion
    const requestedDdd = Number(request.query.ddd.toString()).toString();

    const lengthDdd = requestedDdd.length;

    if (lengthDdd !== 2) {
      throw new BadRequestError({
        message: 'DDD deve conter apenas 2 dígitos',
        type: 'ddd_error',
        name: 'DDD_INVALID',
      });
    }
```